File: Programação Orientada a Objetos/TP1/Source/Model/Pessoa.cpp
/*
    Alunos:
    - Felipe Braz Marques
    - Lucas Chagas Mendes
    - Matheus Peixoto Ribeiro Vieira
    - Nicolas Expedito Lana Mendes
    - Pedro Henrique Rabelo Leão de Oliveira
    - Pedro Morais Fernandes
*/
#include "Pessoa.h"

int Pessoa :: incrementador = 0;

Pessoa :: Pessoa(string nome, Cadastro * cadastro){
    this->id = ++incrementador;
    this->nome = nome;
    this->cadastro = cadastro;
}

Pessoa :: ~Pessoa() {
    delete cadastro;
}

string Pessoa :: getNome() const{
    return nome;
}

void Pessoa :: setNome(string nome){
    this->nome = nome;
}

int Pessoa :: getId() const{
    return id;
}

Cadastro* Pessoa :: getCadastro() const{
    return cadastro;
}

void Pessoa :: setCadastro(Cadastro * cadastro){
    this->cadastro = cadastro;
}

void Pessoa :: serialize(ostream& out) const{
    out << "Nome: " << nome
        << "\nId: " << id
        << "\nLogin: " << cadastro->getLogin() << endl;
}

ostream & operator  << (ostream & out, const Pessoa & obj){
    obj.serialize(out);
    return out;
}

File: Programação Orientada a Objetos/TP2/Trabalho-Pratico/src/Modelo/Pessoa.java
/**
 *
 * @author Felipe Marques, Lucas Chagas, Matheus Peixoto, Nicolas Mendes, Pedro Henrique de
Oliveira, Pedro Morais
 */
package Modelo;

import java.util.Objects;

public class Pessoa {
    private static int contador = 0;
            
    private int id;
    private String nome;
    private String sobreNome;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSobreNome() {
        return sobreNome;
    }

    public void setSobreNome(String sobreNome) {
        this.sobreNome = sobreNome;
    }

    public Pessoa(int id, String nome, String sobreNome) {
        this.id = id;
        this.nome = nome;
        this.sobreNome = sobreNome;
    }
    
    public Pessoa(String nome, String sobreNome) {
        this.id = ++contador;
        this.nome = nome;
        this.sobreNome = sobreNome;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Pessoa other = (Pessoa) obj;
        if (this.id != other.id) {
            return false;
        }
        if (!Objects.equals(this.nome, other.nome)) {
            return false;
        }
        return Objects.equals(this.sobreNome, other.sobreNome);
    }

    @Override
    public String toString() {
        return "Pessoa{" + "id=" + id + ", nome=" + nome + ", sobreNome=" + sobreNome + '}';
    }    
    
}